onnxruntime/core/session/custom_ops.cc

```diff
@@ -377,6 +378,30 @@
   API_IMPL_END
 };
 
+ORT_API_STATUS_IMPL(OrtApis::KernelContext_ParallelFor, _In_ const OrtKernelContext* context, _In_ void (*fn)(void*, size_t), _In_ size_t total, _In_ size_t num_batch, _In_ void* usr_data) {
+  API_IMPL_BEGIN
+  if (context && total && usr_data) {
+    const auto* ctx = reinterpret_cast<const onnxruntime::OpKernelContext*>(context);
+    auto* tp = ctx->GetOperatorThreadPool();
+    if (tp) {
```

Comment:
>if (tp) { [](http://example.com/codeflow?start=4&length=9)

if there is no tp on kernel ctx, should we just fallback to the default execution?

Reply:
Skip tp check since both ThreadPool::TryBatchParallelFor and ThreadPool::TrySimpleParallelFor has that fall-back behavior well-defined.